### PR TITLE
Support TCP flags/seq/ack

### PR DESCRIPTION
**Pull request by snir**
Signed-off-by: Stav Nir <stavnir@microsoft.com>

### CODE REVIEW

File: app/pktgen-cmds.c [hunk 1]
```diff
@@ -279,6 +304,39 @@ pktgen_script_save(char *path)
         fprintf(fd, "range %d dst port max %d\n", i, range->dst_port_max);
         fprintf(fd, "range %d dst port inc %d\n", i, range->dst_port_inc);
 
+        fprintf(fd, "\n");
+        fprintf(fd, "range %d tcp flag clr all\n", i);
+        if (range->tcp_flags & URG_FLAG) {
+            fprintf(fd, "range %d tcp flag set %s\n", i, "urg");
+        }
```
Comment (pktgen) [inline app/pktgen-cmds.c]: For coding style I try not to have single line 'if', 'for', 'while', ... statements with {} around. Can you please remove the {} around these single lines.

File: app/pktgen-cmds.c [hunk 2]
```diff
@@ -3204,6 +3263,266 @@ rate_set_ipaddr(port_info_t *info, char type, struct pg_ipaddr *ip, int ip_ver)
     pktgen_set_tx_update(info);
 }
 
+static uint8_t tcp_flag_from_str(const char *str)
+{
+    if (_cp("urg")) {
+        return URG_FLAG;
```
Comment (pktgen) [inline app/pktgen-cmds.c:3257]: Single line statements.

File: app/pktgen-cmds.c [hunk 3]
```diff
@@ -3204,6 +3263,266 @@ rate_set_ipaddr(port_info_t *info, char type, struct pg_ipaddr *ip, int ip_ver)
...
+ *
+ * SEE ALSO:
+ */
+
+void
+range_set_tcp_seq(port_info_t *info, char *what, uint32_t seq)
+{
+    if (!strcmp(what, "inc") || !strcmp(what, "increment")) {
+        info->range.tcp_seq_inc = seq;
+    } else {
```
Comment (pktgen) [inline app/pktgen-cmds.c]: This one single line one too.

File: app/pktgen-cmds.c [hunk 4]
```diff
@@ -3204,6 +3263,266 @@ rate_set_ipaddr(port_info_t *info, char type, struct pg_ipaddr *ip, int ip_ver)
...
+ *
+ * SEE ALSO:
+ */
+
+void
+range_set_tcp_ack(port_info_t *info, char *what, uint32_t ack)
+{
+    if (!strcmp(what, "inc") || !strcmp(what, "increment")) {
+        info->range.tcp_ack_inc = ack;
+    } else {
```
Comment (pktgen) [inline app/pktgen-cmds.c]: This one too

**Review by pktgen — CHANGES_REQUESTED** [review]
This is excellent work and thank you for submitting your patch.

I only had a few style comments and if you can update the style it look good to go.

**Comment by snir** [comment]
Thanks for the review, fixed it.

**Pull request merged**